drools-osgi/drools-karaf-itests/src/test/java/org/drools/karaf/itest/KieSpringjBPMPersistenceOnKarafKarafIntegrationTest.java

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright 2015 Red Hat
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+package org.drools.karaf.itest;
+
+import org.h2.tools.Server;
+import org.jbpm.process.instance.impl.demo.SystemOutWorkItemHandler;
+import org.jbpm.services.task.identity.JBossUserGroupCallbackImpl;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.kie.api.KieServices;
+import org.kie.api.io.ResourceType;
+import org.kie.api.runtime.EnvironmentName;
+import org.kie.api.runtime.KieSession;
+import org.kie.api.runtime.manager.*;
+import org.kie.api.runtime.process.ProcessInstance;
+import org.kie.api.task.TaskService;
+import org.kie.api.task.model.TaskSummary;
+import org.kie.internal.runtime.manager.context.EmptyContext;
+import org.ops4j.pax.exam.Configuration;
```

Comment:
@triceo  I believe we could. We'd need to:
- enhance PAX-EXAM configuration to be able to use other containers than community Karaf, 
- add a mechanism for configuring other Maven repositories for the OSGi container, (some) OSGi containers do not read repositories settings from Maven settings.xml,
- probably also set up profiles for different versions of the OSGi container (right now we need it, not sure if it will be needed for future versions).